src/geo/LatLng.js

```diff
@@ -28,23 +28,34 @@ import {toLatLngBounds} from './LatLngBounds';
  */
 
 export function LatLng(lat, lng, alt) {
-	if (isNaN(lat) || isNaN(lng)) {
-		throw new Error('Invalid LatLng object: (' + lat + ', ' + lng + ')');
-	}
-
 	// @property lat: Number
 	// Latitude in degrees
-	this.lat = +lat;
+	this.lat = checkNumber(lat);
 
 	// @property lng: Number
 	// Longitude in degrees
-	this.lng = +lng;
+	this.lng = checkNumber(lng);
 
 	// @property alt: Number
 	// Altitude in meters (optional)
-	if (alt !== undefined) {
-		this.alt = +alt;
+	if (arguments.length === 3) {
+		this.alt = checkNumber(alt);
+	}
+}
+
+function checkNumber(a) {

```

Comment:
I would suggest to move this into Utils, maybe we will have some other places where we want to use this. (#7432)

Reply:
I do not agree, because the function implementation is heavily defined by it's particular application.
E.g. we may need integer numbers, or allow Infinity, etc.

Reply:
Hmm you are right but I still think it would be cleaner to have on general function that can used outside of Leaflet core too. Maybe a second optional parameter `allowedTypes` can be passed as array `['integer', 'double', 'Infinity']`.

For infinity it would be also possible to use `if(value === Infinity || checkNumber(value))`

Reply:
> Maybe a second optional parameter `allowedTypes` can be passed as array

Too many parameters to take into account all possible cases. E.g. should it accept hex strings? Exponential form?
BTW, the function itself is not perfect too. I'd like to improve it, but in reasonable limits. May be you have ideas.

Reply:
But we can define / document which cases / types are allowed. 

A way to get `0x10` managed is to check if the string is starting with `0`, length > 1 and the second char is not `.` for decimal numbers. With that `0010` is invalid too.

Reply:
Too much actions. Can be performance penalties